cardea-server/src/main/java/ca/on/oicr/gsi/cardea/server/service/CaseService.java

```diff
@@ -125,6 +123,28 @@ private Set<String> getRunNamesFor(Case kase) {
         .collect(Collectors.toSet());
   }
 
+  public DjerbaCases getDjerbaCases(String requisitionName) {
+    Set<Case> cases = caseData.getCases().stream()
+        .filter(kase -> requisitionName.equals(kase.getRequisition().getName()))
+        .collect(Collectors.toSet());
+    if (cases.isEmpty()) {
+      return null;
+    }
+    Long assayId = cases.stream()
+        .map(kase -> kase.getRequisition().getAssayId())
+        .findFirst().get();
+
+    Assay assay = caseData.getAssaysById().get(assayId);
+    if (cases.isEmpty() || assayId == null) {
```

Comment:
Already checked `cases.isEmpty` above, and it shouldn't be possible for `assayId` to be null

Reply:
Deleted